distil/primitives/satellite_image_loader.py

```diff
@@ -171,13 +179,36 @@ def produce(self, *, inputs: base.FileReaderInputs, timeout: float = None, itera
         file_column_name = inputs_clone.columns[column_index]
         band_column_name = 'band'
 
+        start = time.time()
+        logger.debug('Loading images')
+
         # group by grouping key to get all the images loaded in one row
-        grouped_images = inputs_clone.groupby([grouping_name], sort=False) \
-            .apply(lambda x: self._load_image_group(x[file_column_name], x[band_column_name], base_uri)) \
-            .rename(file_column_name).reset_index(drop=True)
-        grouped_df = container.DataFrame({file_column_name: grouped_images}, generate_metadata=False)
-        grouped_df.metadata = grouped_df.metadata.generate(grouped_df, compact=True)
+        groups = inputs_clone.groupby([grouping_name], sort=False)
+
+        # use the max dimension for the first group as the max dimension for all groups
+        group_key = list(groups.groups.keys())[0] # there must be a better way to do this
```

Comment:
`groups[grouping_name].first()[0]`

This should do it, if it's more preferable